arrows/qt/EmbeddedPipelineWorker.cxx

```diff
@@ -0,0 +1,319 @@
+/*ckwg +29
+ * Copyright 2018 by Kitware, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  * Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ *  * Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ *  * Neither name of Kitware, Inc. nor the names of any contributors may be used
+ *    to endorse or promote products derived from this software without specific
+ *    prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "EmbeddedPipelineWorker.h"
+
+#include <sprokit/processes/adapters/embedded_pipeline.h>
+
+#include <QDebug>
+#include <QEventLoop>
+#include <QFileInfo>
+#include <QThread>
+
+#include <fstream>
+
+namespace kv = kwiver::vital;
+namespace kaq = kwiver::arrows::qt;
+
+using kaq::EmbeddedPipelineWorker;
+using kaq::EmbeddedPipelineWorkerPrivate;
+
+using RequiredEndcaps = kaq::EmbeddedPipelineWorker::RequiredEndcaps;
+
+namespace {
+
+// ----------------------------------------------------------------------------
+std::string
+stdString( QString const& in )
+{
+  auto const& data = in.toLocal8Bit();
+  return std::string{ data.constData(), static_cast< size_t >( data.size() ) };
+}
```

Comment:
Yes. (Well, also because I'm used to Qt4 which used the even-more-wrong `toAscii()`.) AFAIK that is still "correct" on ~~Windows~~ screwy systems in case the system's "native" character set is not UTF-8.

Actually, I see in Qt5 this could be replaced with `toLocal8Bit().toStdString()`...

Reply:
If it can be done that way in Qt5, and if we require Qt5, then I'd rather do that than pass data pointers around.

Reply:
Windows? Screwy? Whatchu talkin' 'bout, Willis? :stuck_out_tongue_winking_eye: